### DISCUSSION TITLE
Remove negative signals

### DISCUSSION THREAD

**Discussion by andreacicconardi94** [hyperspy/hyperspy#3276] [Q&A] [answered]
Dears,
I am working on EELS signals. When I remove the zero-loss peak using the method remove_background, it happens that some values, with respect of the Y axis, are negative (probably because o noise). I want to remove those values, setting them to zero.

For example if in my signal I have something like:

array([0, 11, 3, -2, 2, 7, -1])

The final results that I want is:

array([0, 11, 3, 0, 2, 7, 0])

I think that I can do it extracting the data from the signal and then working on the array, creating a new signal, but I want to know if there is some method that works in place.
Thank you in advance for your help.

Best regards

**Comment by CSSFrancis** [comment, marked as answer]
@andreacicconardi94 This is a good question. 

The easiest way to do this is just adjust the data array in place.

```python
s # Some signal
s.data[s.data<0]=0
```

This should just take any of the data below zero and set it to zero.
